### PR TITLE
9117: Prevent api_key sharing

### LINKED ISSUE #9117: [PYTHON] Apikey mess between Configuration() instances in Python client.
##### Description
I use generated python client with different `apikeys` in different instances. Setting it in one instance changed `apikey` in other instances too. 
It seems an error in creating Configuration object by coping the `default` object, after it all instances shares the same `apikey` dictionary.  

##### Swagger-codegen version

2.4.0

##### Steps to reproduce

- Create 2 instances of Configuration()
- Set apikey in one of them.
- See value from step 2 in all instances and in `default` field

##### Related issues/PRs

https://github.com/swagger-api/swagger-codegen/pull/6554/commits/4677996ae8a292897f92306c64ec8a5bf2cac483

https://github.com/swagger-api/swagger-codegen/pull/6554

##### Suggest a fix/enhancement

As a quick fix I just removed creating Configuration by cloning `default` object and use the constructor directly.

### PR DESCRIPTION
**Pull request by ruslanlazin**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR
These changes prevent sharing `api_key`, `api_key_prefix`, `logger` between configuration instances.
Error was caused by using shallow copy in #6554

Test cases added.

fix #9117

### CODE REVIEW

**Comment by ruslanlazin** [comment]
@kenjones-cisco
review please

File: modules/swagger-codegen/src/main/resources/python/configuration.mustache [hunk 1]
```diff
@@ -22,10 +22,16 @@ class TypeWithDefault(type):
     def __call__(cls):
         if cls._default is None:
             cls._default = type.__call__(cls)
-        return copy.copy(cls._default)
+        return cls._two_level_copy(cls._default)
```
Comment (kenjones-cisco) [inline modules/swagger-codegen/src/main/resources/python/configuration.mustache]: did you try to use `copy.deepcopy(cls._default)` to see if that would provide the expected behavior?

Comment (ruslanlazin) [reply modules/swagger-codegen/src/main/resources/python/configuration.mustache]: Yes, sure. Deepcopy fails with `TypeError: can't pickle thread.lock objects` cause we have reach attributes such as logger file/handler.

**Comment by ruslanlazin** [comment]
@kenjones-cisco 
I've refactored it to make more readable and avoid coping when default configuration is not used.
Tests added.
Please review

**Comment by kenjones-cisco** [comment]
This would definitely be much easier to read and maintain.
LGTM!

**Comment by ruslanlazin** [comment]
@wing328 Seems to be ready fo merge.

**Comment by HugoMario** [comment]
thanks @ruslanlazin @kenjones-cisco , i'll take a time to port these changes to 3.0.0 version.

**Pull request merged**